cpp/src/arrow/util/parsing.h

```diff
@@ -361,6 +361,12 @@ class StringConverter<Int32Type> : public StringToSignedIntConverterMixin<Int32T
 template <>
 class StringConverter<Int64Type> : public StringToSignedIntConverterMixin<Int64Type> {};
 
+using arrow::util::date::day;
```

Comment:
Generally, top-level `using` declarations should only be for `.cc` files. Putting such a declaration in a header file will affect all files including that header files.
(also, in this case it risks produding name collisions, as you've seen below with `year`)

Reply:
Done